### PR TITLE
tines: fix handling of multimorphic tines.audit_log.inputs.inputs.options fields

### LINKED ISSUE #11886: [Tines]: Field mapping issue for FormFieldUpdate event in audit_logs
### Integration Name

Tines [tines]

### Dataset Name

tines.audit_logs

### Integration Version

1.1.0

### Agent Version

NA

### Agent Output Type

elasticsearch

### Elasticsearch Version

8.15.2

### OS Version and Architecture

NA

### Software/API Version

_No response_

### Error Message

object mapping for [tines.audit_log.inputs.inputs.options] tried to parse field [null] as object, but found a concrete value" is observed and the event is dropped. 

### Event Original

```
{"created_at":"2024-11-25T10:35:55Z","id":41274135,"inputs":{"fieldId":1688864,"inputs":{"actionIdsToUpdate":[],"content":"Jac O'All-Trades","dateMode":"DATE","defaultValue":null,"description":"","graphDirection":"HORIZONTAL","graphType":"LINE","maxCharacters":null,"multiSelect":false,"name":"Heading","options":["Option 1","Option 2"],"required":false,"storeValueAs":"","style":"PRIMARY","validationPattern":null}},"operation_name":"FormFieldUpdate","outputs":{},"request_ip":"92.19.2.1","request_user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36","tenant_id":1111,"updated_at":"2024-11-25T10:35:55Z","user_email":"user@customer.com","user_id":58468,"user_name":"First Last"}
```

### What did you do?

Filebeat ingests the Audit logs data and sends the data to the pipeline from the tines integration. 

### What did you see?

The events with `FormFieldUpdate` operation_name are dropped. 

### What did you expect to see?

The data is parsed and indexed. 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] 

## Related issues

- Fixes #11886

### CODE REVIEW

File: packages/tines/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -113,6 +113,52 @@ processors:
...
+      on_failure:
+        - append:
+            field: error.message
+            value: >-
+              Processor '{{{ _ingest.on_failure_processor_type }}}'
+              {{{#_ingest.on_failure_processor_tag}}}with tag '{{{ _ingest.on_failure_processor_tag }}}'
+              {{{/_ingest.on_failure_processor_tag}}}failed with message '{{{ _ingest.on_failure_message }}}'
+        - remove:
+            field: tines.audit_log.inputs.fieldId
+            ignore_failure: true
```
Comment (kcreddy) [inline packages/tines/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml:146]:
```suggestion
        - remove:
            field: tines.audit_log.inputs.inputs.multiSelect
            ignore_failure: true
```
?

File: packages/tines/data_stream/audit_logs/_dev/test/pipeline/test-audit-logs.json [hunk 2]
```diff
@@ -17,6 +17,9 @@
...
+        },
+        {
+            "message": "{\"created_at\":\"2024-11-25T10:35:55Z\",\"id\":41274135,\"inputs\":{\"fieldId\":1688864,\"inputs\":{\"actionIdsToUpdate\":[],\"content\":\"Jac O'All-Trades\",\"dateMode\":\"DATE\",\"defaultValue\":null,\"description\":\"\",\"graphDirection\":\"HORIZONTAL\",\"graphType\":\"LINE\",\"maxCharacters\":null,\"multiSelect\":false,\"name\":\"Heading\",\"options\":[\"Option 1\",\"Option 2\"],\"required\":false,\"storeValueAs\":\"\",\"style\":\"PRIMARY\",\"validationPattern\":null}},\"operation_name\":\"FormFieldUpdate\",\"outputs\":{},\"request_ip\":\"81.2.69.144\",\"request_user_agent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36\",\"tenant_id\":1111,\"updated_at\":\"2024-11-25T10:35:55Z\",\"user_email\":\"user@customer.com\",\"user_id\":58468,\"user_name\":\"First Last\"}"
         }
     ]
 }
```
Comment (ShourieG) [inline packages/tines/data_stream/audit_logs/_dev/test/pipeline/test-audit-logs.json:25]: newline ?

File: packages/tines/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -113,6 +113,52 @@ processors:
...
+      if: ctx.tines?.audit_log?.inputs?.inputs?.required != null
+      on_failure:
+        - append:
+            field: error.message
+            value: >-
+              Processor '{{{ _ingest.on_failure_processor_type }}}'
+              {{{#_ingest.on_failure_processor_tag}}}with tag '{{{ _ingest.on_failure_processor_tag }}}'
+              {{{/_ingest.on_failure_processor_tag}}}failed with message '{{{ _ingest.on_failure_message }}}'
+        - remove:
+            field: tines.audit_log.inputs.fieldId
```
Comment (kcreddy) [inline packages/tines/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml:160]:
```suggestion
            field: tines.audit_log.inputs.inputs.required
```

**Review by kcreddy — APPROVED** [review]
LGTM after suggestion.

File: packages/tines/data_stream/audit_logs/_dev/test/pipeline/test-audit-logs.json (hunk 2 above)
Comment (efd6) [reply packages/tines/data_stream/audit_logs/_dev/test/pipeline/test-audit-logs.json:25]: That was not changed. AFAICS these json files holding message strings never have a final new line>

File: packages/tines/data_stream/audit_logs/fields/fields.yml [hunk 4]
```diff
@@ -37,8 +39,6 @@
                   type: long
                 - name: isServiceToken
                   type: boolean
-                - name: name
```
Comment (ShourieG) [inline packages/tines/data_stream/audit_logs/fields/fields.yml:40]: is this no longer required ?

File: packages/tines/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (efd6) [reply packages/tines/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml]: Also for the one below.

File: packages/tines/data_stream/audit_logs/fields/fields.yml (hunk 4 above)
Comment (efd6) [reply packages/tines/data_stream/audit_logs/fields/fields.yml:40]: Look down.

**Review by ShourieG — APPROVED** [review]

**Pull request merged**